functions/parse.js

```diff
@@ -18,11 +13,6 @@ const parse = (version, options) => {
     return null
   }
 
-  const r = options.loose ? re[t.LOOSE] : re[t.FULL]
-  if (!r.test(version)) {
-    return null
-  }
-
```

Comment:
This is what the `SemVer` constructor is doing.  As you can see it's not even consistent with this code in that `SemVer` `trim()`s the version.
```
    const m = version.trim().match(options.loose ? re[t.LOOSE] : re[t.FULL])

    if (!m) {
      throw new TypeError(`Invalid Version: ${version}`)
    }
```

Reply:
Yep, that's why I removed it, the idea in that test is just to fail-fast.

So, can we consider that a bug or we could try to push this change?

Reply:
I think it's evidence in favor of this change.  We don't need to optimize for bad semver here, and this PR is also a code cleanup and consistency issue.